src/types/json.h

```diff
@@ -134,6 +134,34 @@ struct JsonValue {
     return result_count;
   }
 
+  StatusOr<std::vector<ssize_t>> ArrIndex(std::string_view path, const jsoncons::json &needle, int start,
+                                          int end) const {
+    std::vector<ssize_t> result;
+    try {
+      jsoncons::jsonpath::json_query(value, path, [&](const std::string & /*path*/, const jsoncons::json &val) {
+        if (!val.is_array()) {
+          result.emplace_back(-2);
+          return;
+        }
+        int pstart = std::max<int>(start, 0);
+        int pend = std::max(std::min(static_cast<int>(val.size()), end), 0);
+        auto arr_begin = val.array_range().begin();
+        auto begin_it = arr_begin + pstart;
+
+        auto end_it = arr_begin + pend;
+        auto it = std::find(begin_it, end_it, needle);
+        if (it != end_it) {
+          result.emplace_back(it - arr_begin);
+          return;
+        }
```

Comment:
> Negative values are interpreted as starting from the end.

From the document, I think we need to calculate the real pos while `end` is negative.

Reply:
yes, I have rewrite by normalize array indices as [redis_json](https://github.com/RedisJSON/RedisJSON/blob/master/redis_json/src/redisjson.rs#L40C4-L40C4) does.